### PR TITLE
Feature/api call

### PR DESCRIPTION
**Pull request by MarcelineBall**
# Describe Changes

Please include changes made or which issues were fixed.

# Type of Change
- [ ] Feature HTML
- [x] Feature JavaScript
- [ ] Feature CSS
- [x] Feature React
- [x] Refactor/Fix Bugs

# Please describe the change made

Please give a short description of the changes that were made below
- Refactor:
- Refactored the state to take in the data from our api call
- Refactored the handle click function to make api call for the movie details

- Feature:
- Added api calls file for our api call functions
- Added 500 level error handling
- Added conditional rendering for a loading message, the error message, and the api calls

# How has this been tested?

- [x] In Browser
- [ ] In Console

### CODE REVIEW

**Comment by MarcelineBall** [comment]
@letakeane @sertmer

File: src/App/App.js [hunk 1]
```diff
@@ -1,36 +1,57 @@
...
+      })
+      .catch(() => this.setState(
+        { error: 'Something went wrong, please try again later.' }))
+  }
+
+
   render() {
     return (
       <div className="container">
       <main className='main'>
```
Comment (kendallha) [inline src/App/App.js:45]: The `<main>` needs to be indented by 2 here.

File: src/App/App.js [hunk 2]
```diff
@@ -1,36 +1,57 @@
...
+        {!this.state.selectedMovie && !this.state.error && <>
           <Header />
           <MovieList movies={ this.state.movies } handleClick={this.handleClick}/>
         </>}
+        {!this.state.movies.length && !this.state.error && <>
+          <h1 className='loading-message'>Movies are on the way!</h1>
+          </>}
       </main>
       {this.state.selectedMovie && <MovieDetail movie={this.state.selectedMovie} goBack={this.goBacktoMainView}/>}
       </div>
```
Comment (kendallha) [inline src/App/App.js:57]: We might consider breaking line 56 onto separate lines for easier readability.

File: src/App/App.js [hunk 3]
```diff
@@ -1,36 +1,57 @@
...
+    .then(data => {
+      this.setState({ movies: data.movies })
+    })
+    .catch(() => this.setState(
+      { error: 'Something went wrong, please try again later.' }))
   }
 
   goBacktoMainView = () => {
     this.setState({ selectedMovie: null });
   }
```
Comment (kendallha) [inline src/App/App.js:30]: There is an extra line space here that could be removed.

**Review by kendallha — APPROVED** [review]

**Comment by kendallha** [comment]
This looks great! I like the error messaging and loading message for improved user experience. In a future iteration, we might consider adding more specific error messages depending on the fetch error.

File: src/App/App.css [hunk 4]
```diff
@@ -11,3 +11,9 @@ html, body, .root, .container {
   color: #E6E6E6;
   font-family: 'Farro', sans-serif;
 }
+
+.error-message,
+.loading-message {
```
Comment (letakeane) [inline src/App/App.css:16]: Love this CSS grouping

File: src/App/App.js [hunk 5]
```diff
@@ -1,38 +1,58 @@
...
 import MovieDetail from '../MovieDetail/MovieDetail';
+import { getAllMovies, getMovieDetails } from '../api-calls/api-calls';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
-      selectedMovie: null
+      movies: [],
```
Comment (letakeane) [inline src/App/App.js:12]: Great starting value

File: src/App/App.js [hunk 6]
```diff
@@ -1,38 +1,58 @@
...
+      error:''
     }
   }
 
-  handleClick = (id) => {
-    const clickedMovie = this.state.movies.find(movie => movie.id === id);
-    this.setState({ selectedMovie: clickedMovie });
+  componentDidMount() {
+    getAllMovies()
+    .then(data => {
```
Comment (letakeane) [inline src/App/App.js:20]: The syntax difference between setting movies in state and setting an error in state should probably be made consistent. Either use the implicit return for both or neither.

File: src/App/App.js [hunk 7]
```diff
@@ -1,38 +1,58 @@
...
   goBacktoMainView = () => {
     this.setState({ selectedMovie: null });
   }
 
+  handleClick = (id) => {
+    getMovieDetails(id)
+      .then(data => {
+        this.setState({ selectedMovie: data.movie })
+      })
+      .catch(() => this.setState(
```
Comment (letakeane) [inline src/App/App.js:36]: Same syntax note here - be consistent. Use the implicit return for both, or neither.

File: src/App/App.js [hunk 8]
```diff
@@ -1,38 +1,58 @@
...
       <div className="container">
-      <main className='main'>
-        {!this.state.selectedMovie && <>
-          <Header />
-          <MovieList movies={ this.state.movies } handleClick={this.handleClick}/>
-        </>}
-      </main>
-      {this.state.selectedMovie && <MovieDetail movie={this.state.selectedMovie} goBack={this.goBacktoMainView}/>}
+        <main className='main'>
+          {this.state.error &&
```
Comment (letakeane) [inline src/App/App.js:44]: Great use of conditional rendering syntax

**Review by letakeane — COMMENTED** [review]
Nice work overall! Looks like you've got a good handle on the concepts :) Well done

**Pull request merged**